pkg/controller/quota/quotaforestmanager/qm_lib_backend_with_quotasubt_mgr.go

```diff
@@ -569,9 +570,41 @@ func (qm *QuotaManager) Fits(aw *arbv1.AppWrapper, awResDemands *clusterstateapi
 		return doesFit, preemptIds, strings.TrimSpace(allocResponse.GetMessage())
 	}
 	preemptIds = qm.getAppWrappers(allocResponse.GetPreemptedIds())
+
+	// Update cluster resources in the even that preemption happens
+	if clusterResources != nil {
```

Comment:
I am wondering if this code can be moved from the quota management library and added to the controller code. The proposal is quota management library should only operate on trees. resource checks or whether an AW can run should be the responsibility of the controller code.